components/camel-openapi-java/src/main/java/org/apache/camel/openapi/OpenApiRestProducerFactory.java

```diff
@@ -176,13 +176,13 @@ private Producer createHttpProducer(CamelContext camelContext, OasDocument openA
                 }
                 if (list != null) {
                     for (String s : list) {
-                        csb.append(s);
+                        producesBuilder.add(s);
                     }
                 }
-                produces = csb.isEmpty() ? null : csb.toString();
+                produces = producesBuilder.toString().isEmpty() ? null : producesBuilder.toString();
```

Comment:
I think we can avoid the 2 x toString as that is a costly operation. We can use length method to know if its empty or not. For example if length > 1 (as delimiter is 1) I think. There is a few other places where toString would be called twice like this.

Reply:
The length is 0 by the way, the delimiter is not added. If the value is null, the length gets prefix + suffix, which in this case is empty.